install.sh

```diff
@@ -192,11 +192,61 @@ legacy_transformations_end() {
 ############################################################
 legacy_transformations_repo_end() {
 
+    # Legacy values
     git -C "$1" config --local --unset githooks.autoupdate.registered
 
+    legacy_transformation_adjust_local_paths "$1" || return 1
+
     return 0
 }
 
+############################################################
+# Function to adjust local paths in `.githooks/.shared`
+#   file which are forbidden.
+#
+# Returns:
+#   1 when failed, 0 otherwise
+############################################################
+legacy_transformation_adjust_local_paths() {
+
+    if [ "$(git -C "$1" rev-parse --is-bare-repository 2>/dev/null)" = "true" ]; then
+        SHARED_FILE="$1/.githooks/.shared"
+    else
+        SHARED_FILE="$1/../.githooks/.shared"
+    fi
+
+    if [ -f "$SHARED_FILE" ]; then
+
+        NON_URLS=$(grep -E "^[^#].+$" <"$SHARED_FILE" | grep -vE "^[^/]+://")
+
+        if [ -n "$NON_URLS" ]; then
+
+            echo "! Warning: The shared hooks in \`$(pwd)/.githooks/.shared\` contain" >&2
+            echo "  local paths which are not supported any more:" >&2
+            echo "$NON_URLS" | sed -E "s/^/   - /" >&2
+            echo "  These are now changed to \`file://...\` and the local git config" >&2
+            echo "  \`githooks.failOnNonExistingSharedHooks\` is turned on." >&2
+            echo "  This means you are forced to update the shared hooks in \`$(pwd)\` by running:" >&2
+            echo "    \$ git hooks shared update" >&2
+            echo "  Once updated you can turn \`githooks.failOnNonExistingSharedHooks\` off again by running:" >&2
+            echo "    \$ git hooks config disable fail-on-non-existing-shared-hooks --local" >&2
+
+            git -C "$1" config githooks.failOnNonExistingSharedHooks true >/dev/null 2>&1
+
+            # Add to all non-URLS a `file://` (https://www.gnu.org/software/sed/manual/sed.html#Branching-and-flow-control)
+            # Explanation:
+            # - select first all non-comment lines
+            # - then select the ones which are not starting with <protocol>://
+            # - then adding file:// to them...
+            TEMP=$(mktemp)
+            sed -E '/^[^#].+$/ { /[^\/]+:\/\//! { s/^/file:\/\//g } }' <"$SHARED_FILE" >"$TEMP" &&
```

Comment:
:see_no_evil: 
This code is a bit complicated, and as a user I can't say I'd filly understand what's happening and what I need to do.
Can we not just put a `#` at the start of those paths and tell the user we disabled the unsupported shared repos (perhaps after a prompt) ?

Reply:
Fair enough, but the question still remains, why are these now unsupported... Any suggestion on the warning message?
This fix, turns them into `file://` which is reproduces **exactly** the same behavior for users after updating. Such that their share repos continue to execute.

Reply:
Hm, but how of we don't support local paths there anymore (ecen prefixed with file://), is that not the case?

Reply:
Just uncommenting, is a bit risky... Shall I better explain the sed command? Its just incremental matching lines until the last search/replace

Reply:
We can talk about the sed later, if it has tests I'm OK with it.

My main question is why does this change keep those local paths (now converted to file URLs) working in the .shared file, when we are supposed to refuse running them now?

Reply:
Ah shit, lol, brain mistake, sorry. Jeah, its totaly stupid :-)... ups,

Reply:
We should actually move the to the local `githooks.shared`.

Reply:
If we want perfection...

Reply:
I think that might be pretty easy, just delete them and add them back to the --local `githooks.shared` config. But this feature is in #125. So we probably might include this changeset as well.

Reply:
Ah yeah, that could work nicely.
Any chance we could get that change for the shared config in without any of this here first?

Reply:
I think yes. I just need to rebase, later this or next day

Reply:
Sure, no rush, thanks for all the changes!
Also sorry for being a bit slow with responses, this week is a bit busy...

Reply:
See #125 : This is to much of a burden, and we need a `dev` branch. Because I have the following plan:
- First merge #119 into `dev` 
- Then merge #125 into `dev`
- Then potentially bugfix legacy transform changes in #119
- Then make PR to introduce a `common.sh` script to outsource common functionality (really its high time!)
- Write some additiional tests for these common functions `is_local_path, is_local_url` etc...
- Finalize -> merge back to master

Reply:
So far I introduced the change in this PR as we discussed here.

Reply:
> Then make PR to introduce a common.sh script to outsource common functionality (really its high time!)

I agree it's time, but that should also be it's own PR.
I'm a little bit concerned when I see 4-5 different changes in a single PR because it's 4-5 times more likely to have bugs slip through. 😔